python/across_api/across/resolve.py

```diff
@@ -0,0 +1,170 @@
+    else:
+        return None, None
+
+
+def simbad_radec(name: str) -> Tuple[Optional[float], Optional[float]]:
+    """
+    Given a object name, return the Simbad coordinates in degrees.
+
+    Parameters
+    ----------
+    name : str
+        Name of object to search for
+
+    Returns
+    -------
+    tuple
+        RA/Dec in decimal degrees (float or None)
+    """
+    url = "http://simbad.u-strasbg.fr/simbad/sim-script?script="
+    script = 'format object "%IDLIST(1) | %COO(d;A D)\n' + "query id %s" % name
+
+    lines = requests.get(url + script).text.splitlines()
+
+    ddec = None
+    dra = None
+    for line in lines:
+        x = line.split("|")
+        try:
+            name = x[0]
+            numbers = x[1].split(" ")
+            dra = float(numbers[1])
+            ddec = float(numbers[2].strip())
+        except (ValueError, IndexError):
+            pass
+    return dra, ddec
+
+
+class Resolve(ACROSSAPIBase):
+    """
+    Resolve class for resolving astronomical object names.
+
+    Parameters:
+    -----------
+    name : str
+        The name of the astronomical object to resolve.
+
+    Attributes:
+    -----------
+    status : JobInfo
+        The status of the job.
+    ra : Optional[float]
+        The right ascension of the resolved object.
+    dec : Optional[float]
+        The declination of the resolved object.
+    name : str
+        The name of the astronomical object.
+    resolver : Optional[str]
+        The resolver used for resolving the object.
+
+    Methods:
+    --------
+    get() -> bool:
+        Retrieves the resolved object information.
+    """
+
+    api_name: str = "Resolve"
+    _schema = ResolveSchema
+    _get_schema = ResolveGetSchema
+
+    # Type hints
+    status: JobInfo
+    # Class specific values
+    ra: Optional[float]
+    dec: Optional[float]
+    name: str
+    resolver: Optional[str]
+
+    def __init__(self, name: str):
+        self.status = JobInfo()
+        # Class specific values
+        self.ra = None
+        self.dec = None
+        self.name = name
+        self.resolver = None
+        if self.validate_get():
+            self.get()
+
+    @check_cache
+    @register_job
+    def get(self) -> bool:
+        """
+        Retrieves the RA and Dec coordinates for a given name.
+
+        Returns
+        -------
+        bool
+            True if the name is successfully resolved, False otherwise.
+
+        Raises
+        ------
+        HTTPException
+            If the name couldn't be resolved.
+        """
+        # Make sure the required parameters are given in the correct format
+        if not self.validate_get():
+            return False
+
+        """Do a name search"""
+        # Check against the ANTARES broker
+        if "ZTF" in self.name:
+            ra, dec = antares_radec(self.name)
+            if ra is not None:
+                self.ra, self.dec = ra, dec
+                self.resolver = "ANTARES"
+                return True
```

Comment:
Consider using https://pypi.org/project/antares-client/.

Reply:
There's a reason why I don't. It installs a bunch of huge dependencies like pandas that I otherwise don't need. Given the limitations on size with AWS Lambda, and the fact that I only use this one tiny part of the ANTARES API, this is way leaner.

Reply:
What dependencies are there other than pandas? We are quite likely to acquire a dependency on that anyway.

Reply:
It adds pandas (~60Mb), marshmallow, confluent-kafka and a few other small things. One issue I ran into is that its version of confluent-kafka is required to be 1.70, which conflicts with pygcn which requires >2.2.0. This isn't important for now as I don't use pygcn in the API (yet), however for whatever reason, I cannot get this old version of confluent-kafka to build on either of my dev machines, so it completely breaks my dev environment.

Reply:
The [quota for Lambda function size](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-limits.html#compute-and-storage) is 75 GB and can be adjusted up to terabytes upon request. Avoiding a 60 MB dependency is really premature optimization.

Would you please submit a PR upstream to antares_client to unpin their version of confluent-kafka?

Reply:
Lambda zip-file deployment size is max of ~250Mb (uncompressed), and I hit this easily when doing testing, hence why I made this simple code to call the ANTARES REST API. I'm fine making this change, as it's easy to do (my original code used antares-client before I hit these quota issues). I'll keep the code in my back pocket if we have quota issues.

Reply:
OK, pushed the change in caccb8b.